compassion_nordic_accounting/models/res_partner.py

```diff
@@ -10,71 +10,63 @@
 
 import logging
 
-from stdnum.exceptions import InvalidLength, InvalidFormat, InvalidChecksum, InvalidComponent
-from stdnum.no import fodselsnummer
-from stdnum.se import personnummer
+from stdnum.se import personnummer, orgnr as se_org
+from stdnum.no import fodselsnummer, orgnr as no_org
+from stdnum.dk import cpr, cvr
+from stdnum.fi import veronumero, alv
 
 from odoo import models, api, fields
 from odoo.exceptions import UserError
 
 _logger = logging.getLogger(__name__)
+ERROR_MESSAGE = "SSN (Social Security Number): {err_msg}"
+
+# Library for sweden, norway, denmark, finland
+SSN_CONTRY_FMT_LIST = [personnummer, fodselsnummer, cpr, veronumero]
+VAT_COUNTRY_FMT_LIST = [se_org, no_org, cvr, alv]
 
 
 class ResPartner(models.Model):
     _inherit = "res.partner"
 
     social_sec_nr = fields.Char(string="Social security number")
     _country_id = False
-    # TODO implement ref_social_sec_nr to activate that constaints again
-    # _sql_constraints = [
-    #     ('social_sec_nr_unique',
-    #      'UNIQUE(social_sec_nr)',
-    #      'social security number field needs to be unique'
-    #      )
-    # ]
-
-    def _is_swedish(self, partner_country_id):
-        return partner_country_id == self.env.ref("base.se")
-
-    def _is_norwegian(self, partner_country_id):
-        return partner_country_id == self.env.ref("base.no")
 
     @api.depends('social_sec_nr')
     @api.constrains("social_sec_nr")
```

Comment:
You should not mix a constrains method with a compute one. Make it two distinct methods if needed, I think it's a bad idea to compute values inside a constrains method that is designed to be used only for checking if a given value is valid (look at the Odoo doc if needed).